.github/workflows/test.yml

```diff
@@ -104,60 +92,63 @@ jobs:
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          DRY_RUN: ${{ github.event.inputs.dry-run }}
-          VENDOR_TYPE: ${{ github.event.inputs.vendor-type }}
-          NOTIFY_ID: ${{ github.event.inputs.notify-id }}
-          SOFTWARE_NAME: ${{ github.event.inputs.software-name }}
-          SOFTWARE_VERSION: ${{ github.event.inputs.software-version }}
-          BOT_NAME: ${{ secrets.BOT_NAME }}
-          BOT_TOKEN: ${{ secrets.BOT_TOKEN }}
-          PR_BODY: "Triggerd by ${{ github.event.sender.html_url }} from ${{ github.event.repository.html_url }} on `${{ github.event.ref }}`."
-        run: |
-          echo "[INFO] Dry run '${{ env.DRY_RUN }}'"
-          echo "[INFO] Vendor type '${{ env.VENDOR_TYPE }}'"
-          echo "[INFO] Notify ID '${{ env.NOTIFY_ID }}'"
-          echo "[INFO] Software Name '${{ env.SOFTWARE_NAME }}'"
-          echo "[INFO] Software Version '${{ env.SOFTWARE_VERSION }}'"
-          ve1/bin/behave tests/functional/behave_features/ --tags=version-change --logging-level=WARNING --no-capture --no-color
-
-      - name: Approve PR
-        id: approve_pr
-        if: ${{ steps.check_if_release_pr.outputs.charts_release_branch == 'true' }}
-        uses: hmarr/auto-approve-action@v4
-        with:
-          github-token: ${{ secrets.GITHUB_TOKEN }}
+          echo "Not a release PR from dev to charts, so running only smoke tests"
+          echo "test-tags=smoke" | tee -a $GITHUB_OUTPUT
+  
+  run-tests:
+    # No further pull request author checking done here because
+    # check-contributor gates the needed jobs.
+    name: Run Tests
+    needs:
+    - determine-workflow-conditions
+    if: |
+      needs.determine-workflow-conditions.outputs.run-tests == 'true'
+    uses: ./.github/workflows/behave.yml
+    with:
+      # Default tags to 'full' if test-tags is unset for any reason by the time
+      # we get here.
+      tags: ${{ needs.determine-workflow-conditions.outputs.test-tags || 'full' }}
+      behave-logging-level: WARNING
+      pr-body: "Test triggered by release PR ${{ github.event.pull_request.html_url }}."
+      # checkout parameters passed to ensure we're testing the release content
+      checkout-fetch-depth: 0
+      checkout-repository: ${{ github.event.pull_request.head.repo.full_name }}
+      checkout-ref: ${{ github.event.pull_request.head.sha }}
+    secrets:
+      bot-name: ${{ secrets.BOT_NAME }}
+      bot-token: ${{ secrets.BOT_TOKEN }}
 
-      - name: Merge PR
-        id: merge_pr
-        if: ${{ steps.check_if_release_pr.outputs.charts_release_branch == 'true' }}
-        uses: pascalgn/automerge-action@v0.16.2
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          MERGE_METHOD: squash
-          MERGE_LABELS: ""
+  approve-and-merge:
+    name: Approve and merge
+    needs:
+    - determine-workflow-conditions
+    - run-tests
```

Comment:
Same here wrt needs `check-contributor`.

Actually the same could be asked for the jobs in the release.yml workflow